app/controllers/steps/supporting_evidence_controller.rb

```diff
@@ -3,43 +3,41 @@
 module Steps
   class SupportingEvidenceController < Steps::BaseStepController
     skip_before_action :verify_authenticity_token
-    before_action :supporting_evidence
+    before_action :supporting_evidence, :file_uploader
+
+    SUPPORTED_FILE_TYPES = %w[application/msword application/vnd.openxmlformats-officedocument.wordprocessingml.document
+                              application/rtf image/jpeg image/bmp image/png image/tiff application/pdf].freeze
+
     def edit
       @form_object = SupportingEvidenceForm.build(
         current_application
       )
     end
 
     def create
-      evidence = SupportingEvidence.create(
-        file_name: params[:documents].original_filename,
-        file_type: params[:documents].content_type,
-        file_size: params[:documents].tempfile.size,
-        claim: current_application,
-        file: params[:documents]
-      )
+      unless SUPPORTED_FILE_TYPES.include? params[:documents].content_type
```

Comment:
It's at this point I would suggest moving the logic here into a service to try and keep the controller as small as possible (also helps with testing...

```
def create
  creator = SupportEvidenceCreator.new(claim, document)
  if creator.create
    return_success({ evidence_id: evidence.id, file_name: params[:documents].original_filename })
  else
    return_error(e, { message: creator.error_message })
  end
end

Reply:
Given the size of this PR already, this can be future enhancement